test/foundry/dapp/Application.t.sol

```diff
@@ -497,6 +529,207 @@ contract ApplicationTest is ERC165Test {
         _appContract.executeOutput(output, proof);
     }
 
+    function testEtherTransferToENS() external _resolveENS {
+        string memory name = "EtherToENSDelegateCallVoucher";
+        bytes memory output = _getOutput(name);
+        OutputValidityProof memory proof = _getProof(name);
+
+        // test revert
+
+        assertLt(
+            address(_appContract).balance,
+            _transferAmount,
+            "Application contract does not have enough ethers"
+        );
+
+        vm.expectRevert();
+        _appContract.executeOutput(output, proof);
+
+        // test success
+
+        vm.deal(address(_appContract), _transferAmount);
+
+        uint256 recipientBalance = address(_recipient).balance;
+        uint256 appBalance = address(_appContract).balance;
+
+        _expectEmitOutputExecuted(output, proof);
+        _appContract.executeOutput(output, proof);
+
+        assertEq(
+            address(_recipient).balance,
+            recipientBalance + _transferAmount,
+            "Recipient should have received the transfer amount"
+        );
+
+        assertEq(
+            address(_appContract).balance,
+            appBalance - _transferAmount,
+            "Application contract should have the transfer amount deducted"
+        );
+
+        assertTrue(
+            _wasOutputExecuted(proof),
+            "Output should be marked as executed"
+        );
+
+        _expectRevertOutputNotReexecutable(output);
+        _appContract.executeOutput(output, proof);
+    }
+
+    function testERC20TransferToENS() external _resolveENS {
+        string memory name = "ERC20ToENSDelegateCallVoucher";
+        bytes memory output = _getOutput(name);
+        OutputValidityProof memory proof = _getProof(name);
+
+        _testERC20Fail(output, proof);
+
+        vm.mockCall(
+            address(_ens),
+            abi.encodeCall(_ens.resolver, (_ensNode)),
+            abi.encode(_resolver)
+        );
+        vm.mockCall(
+            address(_resolver),
+            abi.encodeWithSignature("addr(bytes32)", (_ensNode)),
+            abi.encode(_recipient)
+        );
+
+        _testERC20Success(output, proof);
+    }
+
+    function testERC721TransferToENS() external _resolveENS {
+        string memory name = "ERC721ToENSDelegateCallVoucher";
+        bytes memory output = _getOutput(name);
+        OutputValidityProof memory proof = _getProof(name);
+
+        _testERC721(output, proof);
+    }
+
+    function testERC1155SingleTransferToENS() external _resolveENS {
```

Comment:
I feel the lack of regular ERC-1155 transfer tests, without ENS.
Tip: define internal functions to avoid code duplication.

Reply:
Agree, let's do that in another issue.